state/state.go

```diff
@@ -126,6 +135,56 @@ func (s *State) LoadABCIResponses() *ABCIResponses {
 	return abciResponses
 }
 
+// LoadValidators loads the ValidatorSet for a given height.
+func (s *State) LoadValidators(height int) (*types.ValidatorSet, error) {
+	v := s.loadValidators(height)
+	if v == nil {
+		return nil, ErrNoValSetForHeight{height}
+	}
+
+	if v.ValidatorSet == nil {
+		v = s.loadValidators(v.LastHeightChanged)
```

Comment:
shouldn't we always do this?

Reply:
oh nevermind.  I think rather, maybe if s.LastHeightValidatorsChanged <= height, we can just load from s.LastHeightValidatorsChanged.

Reply:
I think that's right, but it would also be racy, since this function is called by RPC, and s.LastHeightValidatorsChanged is updated by consensus. We'd have to add mutexes to the state - do we want to do that? I think we have an [open issue](https://github.com/tendermint/tendermint/issues/165) for it